### PR TITLE
82 docs add swaggerdocs to traffic sniffer controllers

### PR DESCRIPTION
**Pull request by idodav**
- feat: swagger jsdoc documentation
- feat: swagger jsdoc

### CODE REVIEW

**Comment by idodav** [comment]
Swagger can be accessed through http://localhost:5012/api-docs


Screenshot 2023-07-08 at 22 59 55
Screenshot 2023-07-08 at 23 00 08

File: traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts [hunk 1]
```diff
@@ -127,6 +254,30 @@ export class MockManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     requestBody:
+     *        description: Activate a mock
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts:274]: Is the request body mandatory in this doc? I mean - all the function body references is the params object

File: traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts [hunk 2]
```diff
@@ -105,6 +208,30 @@ export class MockManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     requestBody:
+     *        description: Deactivate mock manager
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts:228]: Is it relevant?

File: traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts [hunk 3]
```diff
@@ -83,6 +162,30 @@ export class MockManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     requestBody:
+     *        description: Activate mock manager
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts:182]: Is it relevant?

File: traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts [hunk 4]
```diff
@@ -61,6 +116,30 @@ export class MockManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     requestBody:
+     *        description: Create a new mock
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts:136]: Missing mockId property

File: traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts [hunk 5]
```diff
@@ -39,6 +70,30 @@ export class MockManagerController {
...
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts:90]: What is the structure of the body?

File: traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts [hunk 6]
```diff
@@ -154,6 +305,30 @@ export class MockManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     requestBody:
+     *        description: Deactivate a mock
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/mock-manager-controller.ts:325]: In the body, mockId is missing

File: traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts [hunk 7]
```diff
@@ -26,6 +52,19 @@ export class SnifferManagerController {
...
+     *   get:
+     *     tags:
+     *      - sniffer
+     *     description: Get a sniffers
+     *     responses:
+     *       200:
+     *         description: Returns a sniffer
+     *       500:
+     *         description: Server error
+     */
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts:67]: Missing reference to parameters object

File: traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts [hunk 8]
```diff
@@ -37,18 +76,55 @@ export class SnifferManagerController {
...
+     *   post:
+     *     tags:
+     *      - sniffer
+     *     description: Create a sniffer
+     *     requestBody:
+     *        description: Create a sniffer
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts:91]: What is the structure of the body?

File: traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts [hunk 9]
```diff
@@ -73,11 +149,28 @@ export class SnifferManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     responses:
+     *       200:
+     *         description: Sniffer started
+     *       500:
+     *         description: Server error
+     */
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts:169]: Structure body is missing

File: traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts [hunk 10]
```diff
@@ -98,6 +191,30 @@ export class SnifferManagerController {
...
+     *       - name: port
+     *         in: query
+     *         description: service port
+     *         required: true
+     *     requestBody:
+     *        description: Execute a request from a sniffer
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts:211]: Add the structure of the body

File: traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts [hunk 11]
```diff
@@ -140,6 +275,30 @@ export class SnifferManagerController {
...
+     *         in: query
+     *         description: service id
+     *         required: true
+     *     requestBody:
+     *        description: Edit a sniffer
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
+     *     responses:
```
Comment (nkotzrenko) [inline traffic-sniffer/lib/sniffer-manager/sniffer-manager-controller.ts:296]: Add structure to body

**Review by nkotzrenko — CHANGES_REQUESTED** [review]
Some missing data

**Review by nkotzrenko — APPROVED** [review]

**Comment by nkotzrenko** [comment]
I approved these changes after speaking to @idodav - we can fix it later (as part of the zod validations PR) on my end

**Pull request merged**